Enable Marvell ethernet switch for Bluefin devices

HatsyRei opened the pull request:
WI: [AB#3033955](https://ni.visualstudio.com/DevCentral/_workitems/edit/3033955)

Bluefin ethernet interface is exposed via a Marvell 88E6341 'Topaz' ethernet switch. These commits pull in the mv88e6xxx driver by enabling `CONFIG_NET_DSA_MV88E6XXX` and its PTP support via `CONFIG_NET_DSA_MV88E6XXX_PTP`. Device tree bindings are added to Bluefin device tree to enable probe and configuration of the switch at runtime. 

Testing: Booted kernel with changes on cDAQ-9189, verified functionality of ethernet ports, sanity check output of ifconfig and content of /sys/class/ptp.

HatsyRei committed:
nati_bluefin_defconfig: Enable Marvell 88E6xxx driver

Bluefin ethernet interface is exposed via a Marvell 88E6341 'Topaz'
ethernet switch. Enable the mv88e6xxx driver along with PTP support.

Signed-off-by: HatsyRei <yan.huan.chng@emerson.com>

chaitu236 commented on arch/arm/configs/nati_bluefin_defconfig:
```diff
@@ -60,7 +60,6 @@ CONFIG_BRIDGE_EBT_MARK_T=y
 CONFIG_BRIDGE=y
 CONFIG_BRIDGE_VLAN_FILTERING=y
 CONFIG_NET_DSA=y
-CONFIG_NET_DSA_MV88E6XXX_ONLY_8021AS=y
```
I assume this got dropped because of `savedefconfig`? Does that mean you don't require the https://github.com/ni/linux/commit/295c911ba097730c1f3fc0974145b50bc5e414bc cherrypick? I was wondering if you needed that when doing 6.12 [rebase](https://github.com/ni/linux/pull/211).

HatsyRei replied:
Yes, these are the results of `savedefconfig`. My understanding is that the 6.6 branch already has upstream DSA changes which obviates the need for https://github.com/ni/linux/commit/295c911ba097730c1f3fc0974145b50bc5e414bc, and it got included in https://github.com/ni/linux/pull/188 due to an oversight. Should I revert this particular commit?

chaitu236 replied:
Ok, I think this is fine.

HatsyRei committed:
ARM: dts: ni-bluefin: Add Marvell ethernet switch bindings

Add appropriate device tree bindings to enable probe and configuration
of Marvell 88E6341 'Topaz' ethernet switch on Bluefin hardware.

Signed-off-by: HatsyRei <yan.huan.chng@emerson.com>

HatsyRei commented on arch/arm/boot/dts/xilinx/ni-bluefin.dts:
```diff
@@ -75,6 +72,55 @@
+			port@0 {
+				reg = <0>;
+				label = "cpu";
+				phy-mode = "rgmii-id";
```
Fixed an issue with phy-mode specification. Now using "rgmii-id" (rgmii with internal delays) as that appears to be the correct configuration for switch on Bluefin, based on https://github.com/ni/linux/commit/8593a737dd13f343e7e4cb8f51919d5cee94bd27. Verified network connectivity after deploying changes on 9189.
